pyanaconda/ui/lib/help.py

```diff
@@ -19,133 +19,165 @@
     #   and only consider it fit to use if it does have the file
-    if not os.path.exists(help_folder):
-        log.warning("help folder %s for help file %s does not exist", help_folder, help_file)
+    if not os.path.exists(help_directory):
+        log.warning("help folder %s for help file %s does not exist", help_directory, help_file)
         return None
 
-    # Collect languages and files that provide the help content.
-    help_langs = {}
-
-    for lang in os.listdir(help_folder):
-        # Does the help file exist for this language?
-        path = os.path.join(help_folder, lang, help_file)
-        if not os.path.isfile(path):
-            continue
-
-        # Create a valid langcode. For example, use en_US instead of en-US.
-        code = lang.replace('-', '_')
-        help_langs[code] = path
-
-    # Find the best help file.
-    for locale in (current_lang, DEFAULT_LANG):
-        best_lang = find_best_locale_match(locale, help_langs.keys())
-        best_path = help_langs.get(best_lang, None)
-
-        if best_path:
-            return best_path
-
-    # No file found.
-    log.warning("no help content found for file %s", help_file)
-    return None
+    help_langs = [l for l in os.listdir(help_directory) if os.path.isfile(os.path.join(help_directory, l, help_file))]
 
+    best_lang = find_best_locale_match(current_lang, help_langs)
+    if not best_lang and current_lang != DEFAULT_LANG:
+        # nothing found for current language, fallback to the default language,
+        # if available & different from current language
+        best_lang = find_best_locale_match(DEFAULT_LANG, help_langs)
 
-def get_help_path(help_file, plain_text=False):
-    """
-    Return the full path for the given help file name,
-    if the help file path does not exist a fallback path is returned.
-    There are actually two possible fallback paths that might be returned:
-
-    * first we try to return path to the main page of the installation guide
-      (if it exists)
-    * if we can't find the main page of the installation page, path to a
-      "no help found" placeholder bundled with Anaconda is returned
-
-    Regarding help l10n, we try to respect the current locale as defined by the
-    "LANG" environmental variable, but fallback to English if localized content
-    is not available.
-
-    :param help_file: help file name
-    :type help_file: str or NoneType
+    # did we get something usable ?
+    if best_lang:
+        # we already checked that the full path exists when enumerating suitable
+        # help content above, so we can just return the part here without
+        # checking it again
+        return os.path.join(help_directory, best_lang, help_file)
+    else:
+        log.warning("no help content found for file %s", help_file)
+        return None
 
-    :param plain_text: should we find the help in plain text?
-    :type plain_text: bool
+def get_placeholder_path(plain_text=False):
```

Comment:
Please, rename it to `get_default_help_file`.

Reply:
Shortened.